### PR TITLE
[libc][docs] Update docs to reflect new headergen

### PR DESCRIPTION
**Pull request by michaelrj-google**
Since new headergen is now the default for building LLVM-libc, the docs
need to be updated to reflect that. While I was editing those docs, I
took a quick pass at updating other out-of-date pages.

### CODE REVIEW

**Review by jhuber6 — COMMENTED** [review]
There should probably be a mention that `pyyaml` needs to be installed.

**Comment by michaelrj-google** [comment]
I've added a note to the fullbuild pages

**Review by aaryanshukla — COMMENTED** [review]
clear conflicts

**Review by jhuber6 — APPROVED** [review]

**Review by aaryanshukla — APPROVED** [review]
LGTM

**Review by lntue — CHANGES_REQUESTED** [review]

File: libc/docs/full_host_build.rst [hunk 1]
```diff
@@ -8,27 +8,97 @@ Full Host Build
...
+   this command:
+   ``sudo ln -s /usr/include/<HOST TRIPLE>/asm /usr/include/asm``
+   (your host triple will probably be similar to ``x86_64-linux-gnu``)
+
+.. code-block:: sh
+
+   $> cd llvm-project  # The llvm-project checkout
+   $> mkdir build
+   $> cd build
+   $> cmake ../runtimes \
```
[the remark this answers is not in the document]
Comment (jhuber6) [reply libc/docs/full_host_build.rst:45]: Hm, we might need to refine the  weird hacky logic for `set(NEED_LIBC_HDRGEN FALSE)`

[the remark this answers is not in the document]
Comment (michaelrj-google) [reply libc/docs/full_host_build.rst:45]: That's odd, when I run it on a fresh build folder it works fine. Could you make sure that you didn't have a lingering CMakeCache.txt and try it again?

[the remark this answers is not in the document]
Comment (lntue) [reply libc/docs/full_host_build.rst:45]: I started with a clean build folder, just sync to the current LLVM head.

[the remark this answers is not in the document]
Comment (michaelrj-google) [reply libc/docs/full_host_build.rst:45]: Ah, the error is showing up now that I've sync'd, I have a guess as to what happened.

[the remark this answers is not in the document]
Comment (michaelrj-google) [reply libc/docs/full_host_build.rst:45]: This patch fixes the issue: https://github.com/llvm/llvm-project/pull/104522
The `elf` and `link` headers were not properly set up to use newheadergen.

[the remark this answers is not in the document]
Comment (michaelrj-google) [reply libc/docs/full_host_build.rst:45]: Now that that patch is landed, could you try the build again? It's passing on my end.

[the remark this answers is not in the document]
Comment (lntue) [reply libc/docs/full_host_build.rst:45]: It works for me now.

File: libc/docs/full_host_build.rst [hunk 2]
```diff
@@ -8,27 +8,97 @@ Full Host Build
...
+
+Build and test
+==============
+
+After configuring the build with the above ``cmake`` command, one can build test
+libc with the following command:
+
+.. code-block:: sh
+
+   $> ninja libc check-libc
```
Comment (lntue) [inline libc/docs/full_host_build.rst:68]: right now in full build `libm.a` is built separately with `ninja libm`.  You might want to add it here, until we combine their cmake targets for full build similar to overlay mode.

**Comment by michaelrj-google** [comment]
I've addressed the comments, and the build works on my system. I'm planning to land this patch tomorrow.

**Review by lntue — APPROVED** [review]

**Pull request merged**